src/react/utils.js

```diff
@@ -33,6 +34,17 @@ export function isReactComponent(name: string) {
   return name.length > 0 && name[0] === name[0].toUpperCase();
 }
 
+export function valueIsClassComponent(realm: Realm, value: Value) {
+  if (!(value instanceof FunctionValue)) {
+    return false;
+  }
+  let prototype = Get(realm, value, "prototype");
+  if (prototype instanceof ObjectValue) {
+    return prototype.properties.has("render");
```

Comment:
Checking `isReactComponent` is a bit safer.
https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactFiber.js#L224

Reply:
I actually saw that same function and replicated here. This is doing the same thing as under the hood as the original logic?

Reply:
This seems to look for `render`. The original function is checking for `isReactComponent`.